src/core_editor/line_buffer.rs

```diff
@@ -454,6 +454,92 @@ impl LineBuffer {
     pub fn is_cursor_at_last_line(&self) -> bool {
         !self.get_buffer()[self.offset()..].contains('\n')
     }
+
+    /// Finds index for the first occurrence of a char to the right of offset
+    pub fn find_char_right(&self, c: &char) -> Option<usize> {
+        if self.offset() + 1 > self.lines.len() {
+            return None;
+        }
+
+        let search_str = &self.lines[self.offset() + 1..self.lines.len()];
+        search_str.find(*c).map(|index| index + self.offset() + 1)
+    }
+
+    /// Finds index for the first occurrence of a char to the left of offset
+    pub fn find_char_left(&self, c: &char) -> Option<usize> {
+        if self.offset() + 1 > self.lines.len() {
+            return None;
+        }
+
+        let search_str = &self.lines[..self.offset()];
+        search_str.rfind(*c)
+    }
+
+    /// Moves the insertion point until the next char to the right
+    pub fn move_right_until(&mut self, c: &char) -> usize {
+        if let Some(index) = self.find_char_right(c) {
+            self.insertion_point.offset = index
+        }
+
+        self.insertion_point.offset
+    }
+
+    /// Moves the insertion point before the next char to the right
+    pub fn move_right_before(&mut self, c: &char) -> usize {
+        if let Some(index) = self.find_char_right(c) {
+            self.insertion_point.offset = index - 1
```

Comment:
Bit tricky. 
We have to also make sure that we end up with the cursor on valid graphemes (e.g. Emojis multiple code points for a single column on the display)
```suggestion
            self.insertion_point.offset = index;
            self.insertion_point.offset = self.grapheme_left_index();
```